### PR TITLE
Link to gnomAD gene SVs v2.1 for SV variants with gnomAD frequency

### LINKED ISSUE #3788: Link to gnomAD SV from SV variant page
Seems the link is only produced for SNVs!

### PR DESCRIPTION
**Pull request by northwestwitch**
FIx #3788 

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. Fist book your testing time using the Pax software available at [https://pax.scilifelab.se/](https://pax.scilifelab.se). The resource you are going to call dibs on is `scout-stage` and the server is `cg-vm1`.
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
1. After testing is done, repeat procedure at [https://pax.scilifelab.se/](https://pax.scilifelab.se), which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

<details>
<summary>Testing on hasta server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. `ssh <USER.NAME>@hasta.scilifelab.se`
1. Book your testing time using the Pax software. `us; paxa -u <user> -s hasta -r scout-stage`. You can also use the WSGI Pax app available at [https://pax.scilifelab.se/](https://pax.scilifelab.se).
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `conda activate S_scout; pip freeze | grep scout-browser`
1. Deploy the branch to test: `bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_scout -t scout -b <this_branch>`
1. Make sure the branch is deployed: `us; scout --version`
1. After testing is done, repeat the `paxa` procedure, which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

**How to test**:
1. Test it with a SV variant which has gnomad frequency in scout (example: https://scout-stage.scilifelab.se/cust000/F0009130-mip7/sv/variants/17ffc4567a660efab974bbd32bbe7d33)

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [x] code approved by DN
- [x] tests executed by CR, DN

### CODE REVIEW

**Comment by northwestwitch** [comment]
What did I just do??  🤣 
image

**Comment by dnil** [comment]
> What did I just do?? 🤣
> image

My thoughts exactly, but it wasn't actually you:
Screenshot 2023-02-01 at 08 13 22

File: scout/server/links.py [hunk 1]
```diff
@@ -101,6 +102,16 @@ def gnomad_str_gene(hgnc_symbol):
     return link.format(hgnc_symbol)
 
 
+def gnomad_sv_gene(hgnc_symbol):
```
Comment (northwestwitch) [inline scout/server/links.py:105]: We don't need this one no? I confused variant and gene links

Comment (dnil) [reply scout/server/links.py:105]: Right, I dont think so!

File: scout/server/links.py [hunk 2]
```diff
@@ -538,6 +547,32 @@ def gnomad_link(variant_obj, build=37):
...
+def gnomad_sv_link(variant_obj, build=37):
+    """Compose link to gnomAD website for a structural variant.
+
+    GnomAD SVs are not yet public for hg38 or MT as 2023-02-01.
+
+    Since we do not track the GnomAD variant ID we link to the region view instead.
+    For balanced variants, we pick the region for the start position (the var is available in both).
+    """
+
+    if build == 38 or variant_obj["chromosome"] in ["M", "MT"]:
```
Comment (northwestwitch) [inline scout/server/links.py:559]: why not returning none?

Comment (dnil) [reply scout/server/links.py:559]: I was thinking that page explains why there is not connection, but either works!

Comment (northwestwitch) [reply scout/server/links.py:559]: I'm thinking that perhaps there won't be any gnomAD frequency if the relative page doesn't exist in gnomAD

File: scout/server/links.py (hunk 1 above)
Comment (northwestwitch) [reply scout/server/links.py]: Removed

File: scout/server/links.py [hunk 3]
```diff
@@ -538,6 +536,32 @@ def gnomad_link(variant_obj, build=37):
...
+        return "http://gnomad.broadinstitute.org/"
+
+    url_template = (
+        "http://gnomad.broadinstitute.org/region/{this[chromosome]}-{this[position]}"
```
Comment (northwestwitch) [inline scout/server/links.py:552]:
```suggestion
        "https://gnomad.broadinstitute.org/region/{this[chromosome]}-{this[position]}"
```

Comment (dnil) [reply scout/server/links.py:552]: Oh, didn't I push that? One moment..

**Comment by dnil** [comment]
Looks good on stage:

Screenshot 2023-02-01 at 10 16 52

Screenshot 2023-02-01 at 10 17 03

**Comment by northwestwitch** [comment]
Another example on stage: https://scout-stage.scilifelab.se/cust000/SWEDAC-2019-HG-05-Trio-NovaSeq/sv/variants/7f3a713e209b1c3b16c7518e083a28a0

**Comment by dnil** [comment]
By the looks of it, we are pointing to roughly the right place on GnomAD. The coordinates are allowed to be off a bit, its SV matching after all. There is increasing concern svdb is not doing quite the right thing with the GnomAD matching, but this has not necessarily been seen on the latest versions of references and software, only on random stage case variants.

File: CHANGELOG.md [hunk 4]
```diff
@@ -15,12 +15,14 @@ About changelog [here](https://keepachangelog.com/en/1.0.0/)
...
 - Added Manual Ranks Risk Factor, Likely Risk Factor and Uncertain Risk Factor
 - Display matching manual ranks from previous cases the user has access to on VariantS and Variant pages
+- Link to gnomAD gene SVs v2.1 for SV variants with gnomAD frequency
 ### Changed
 - Display chrY for sex unknown
 - Deprecate legacy scout_load() method API call.
 - Message shown when variant tag is updated for a variant
 - When all ACMG classifications are deleted from a variant, the current variant classification status is also reset.
 - Refactored the functions that collect causative variants
+- Removed `scripts/generate_test_data.py`
```
Comment (dnil) [inline CHANGELOG.md:25]: 💯

**Review by dnil — APPROVED** [review]

**Pull request merged**